src/main/java/com/google/devtools/build/lib/remote/blobstore/http/AbstractHttpHandler.java

```diff
@@ -78,6 +81,10 @@ protected void addCredentialHeaders(HttpRequest request, URI uri) throws IOExcep
     }
   }
 
+  protected void addUserAgentHeader(HttpRequest request) {
+    request.headers().set(HttpHeaderNames.USER_AGENT, "Bazel/" + BlazeVersionInfo.instance().getReleaseName());
```

Comment:
how about using `getVersion()` instead of `getReleaseName()` e.g. `"bazel " + BlazeVersionInfo.instance().getVersion()`.
This will produce `bazel 0.23` for the 0.23 release and `bazel HEAD` for a development version.

Reply:
Also, since it's a static value consider storing this string in a field, in order to avoid creating a new string for each request.

Reply:
Actually, according to the code the `getVersion()` call returns an empty String for a development version. Other that that, 2 calls are using the same data source - `BUILD_LABEL`